zeus/run/dataloader.py

```diff
@@ -741,46 +730,31 @@ def _is_train(self) -> bool:
         """Return whether this dataloader is for training."""
         return self.split == "train"
 
+    @property
+    def _power_log_prefix(self) -> str:
+        """Build the prefix for the power monitor log file."""
+        return f"{self.logdir}/bs{self.train_batch_size}+e{self.epoch_num}"
+
     def _power_log_path(self, rank: int) -> str:
         """Build the path for the power monitor log file at the GPU with rank."""
         return f"{self.logdir}/bs{self.train_batch_size}+e{self.epoch_num}+gpu{rank}.power.log"
 
-    def _start_monitor(self) -> None:
-        """Start the power monitor subprocess."""
-        # Sanity checks.
-        assert not ZeusDataLoader.monitors
-        # Only the master process starts the monitors.
-        assert self.rank == 0
-
-        # Start monitors. Master process starts and records all monitors.
-        for index in range(self.world_size):
-            monitor = subprocess.Popen(
-                [
-                    self.monitor_path,
-                    self._power_log_path(index),
-                    "0",
-                    "100",
-                    str(index),
-                ],
-            )
-            self._log(f"[GPU_{index}] Zeus monitor started.")
-            ZeusDataLoader.monitors.append(monitor)
+    @property
+    def _get_prof_service(self) -> ZeusProfilingService:
+        """Return the `ZeusProfilingService` instance."""
+        assert ZeusDataLoader.prof_service is not None, f"{self.prof_service=}"
+        return ZeusDataLoader.prof_service
 
-    def _kill_monitor(self) -> None:
-        """Kill the power monitor subprocess."""
-        # Sanity checks.
-        assert ZeusDataLoader.monitors
-        # Only the master process kills the monitors.
+    def _prof_window_push(self) -> None:
+        """A wrapper function that pushes profile window."""
         assert self.rank == 0
+        self._get_prof_service.push_window()
 
-        # Kill monitors.
-        for monitor in ZeusDataLoader.monitors:
-            monitor.send_signal(signal.SIGINT)
-        for monitor in ZeusDataLoader.monitors:
-            monitor.wait(timeout=1.0)
-
-        # Cleanup the monitor list
-        ZeusDataLoader.monitors = []
+    def _prof_window_pop(self) -> Tuple[float, List[float]]:
+        """A wrapper function that pops profile window and stores the profiling result,
+        so that both train and eval dataloader will be able to access the result."""
```

Comment:
Function docstrings usually begin with one sentence in one line that describes its functionality only (and not its implementation details). Then, after a newline, you can add more explanation.